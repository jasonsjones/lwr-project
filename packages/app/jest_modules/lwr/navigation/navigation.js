export function CurrentPageReference() {
    /* empty body */
}

export function NavigationContext() {
    /* empty body */
}

export function navigate() {
    /* empty body */
}

export function generateUrl() {
    /* empty body */
}
