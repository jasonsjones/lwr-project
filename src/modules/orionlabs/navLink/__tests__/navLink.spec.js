import { createElement } from 'lwc';
import { CurrentPageReference } from 'lwr/navigation';
import NavLink from 'orionlabs/navLink';

const pages = {
    home: {
        type: 'home',
        attributes: {},
        state: {}
    },
    about: {
        type: 'namedPage',
        attributes: {
            pageName: 'about'
        },
        state: {}
    }
};

jest.mock('lwr/navigation', () => {
    return {
        NavigationContext: require('@salesforce/wire-service-jest-util').createTestWireAdapter(),
        CurrentPageReference: require('@salesforce/wire-service-jest-util').createTestWireAdapter(),
        navigate: jest.fn(),
        generateUrl: jest.fn()
    };
});

async function createComponent(prop = {}) {
    const element = createElement('orionlabs-nav-link', {
        is: NavLink
    });

    Object.assign(element, prop);
    document.body.appendChild(element);
    await Promise.resolve();

    return element;
}

describe('NavLink component', () => {
    afterEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders home link with active styles', async () => {
        const element = await createComponent({ label: 'Home', pageReference: pages.home });
        CurrentPageReference.emit(pages.home);
        await Promise.resolve();
        const anchorEl = element.shadowRoot.querySelector('a');

        expect(anchorEl.classList).toContain('border-b-2');
        expect(element).toMatchSnapshot();
    });

    it('renders about link without active styles', async () => {
        const element = await createComponent({ label: 'About', pageReference: pages.about });
        CurrentPageReference.emit(pages.home);
        await Promise.resolve();
        const anchorEl = element.shadowRoot.querySelector('a');

        expect(anchorEl.classList).not.toContain('border-b-2');
        expect(element).toMatchSnapshot();
    });
});
