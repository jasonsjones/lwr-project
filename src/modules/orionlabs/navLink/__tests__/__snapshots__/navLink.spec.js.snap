// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component renders 1`] = `
<orionlabs-nav-link>
  #shadow-root(open)
    <a
      class="text-xl hover:text-white"
    />
</orionlabs-nav-link>
`;
